ddl/ddl_api.go

```diff
@@ -1369,6 +1372,109 @@ func checkPartitionByRangeColumn(ctx sessionctx.Context, tbInfo *model.TableInfo
 	return checkAddPartitionTooManyPartitions(uint64(len(pi.Definitions)))
 }
 
+func checkRangeColumnsPartitionType(tbInfo *model.TableInfo, columns []model.CIStr) error {
+	for _, col := range columns {
+		colInfo := getColumnInfoByName(tbInfo, col.L)
+		if colInfo == nil {
+			return errors.Trace(ErrFieldNotFoundPart)
+		}
+		// The permitted data types are shown in the following list:
+		// All integer types
+		// DATE and DATETIME
+		// CHAR, VARCHAR, BINARY, and VARBINARY
+		// See https://dev.mysql.com/doc/mysql-partitioning-excerpt/5.7/en/partitioning-columns.html
+		switch colInfo.FieldType.Tp {
```

Comment:
Why not use `if` here?

Reply:
Because there are many conditions here, the code is more ugly use if:

```
if colInfo.FieldType.Tp == mysql.TypeTiny ||
   colInfo.FieldType.Tp == xxx ||
   colInfo.FieldType.Tp == yyy ||
   colInfo.FieldType.Tp == zzz ||
   ...
```

You see, `colInfo.FieldType.Tp` repeat so many times!